infrastructure/ansible/roles/nginx/tasks/main.yml

```diff
@@ -3,7 +3,16 @@
     name: nginx
     state: present
 
+- name: copy dhparam.pem
+  when: not nginx_generate_dhparam
+  copy:
+    src: vagrant-only.dhparam.pem
+    dest: "{{ dhparams_pem_file }}"
+    owner: root
+    mode: "0600"
+
 - name: generate dhparam.pem
+  when: nginx_generate_dhparam
   shell:
     cmd: openssl dhparam -out {{ dhparams_pem_file }} 4096
     creates: "{{ dhparams_pem_file }}"
```

Comment:
I'm a bit late to the party, but is the when clause required here, or does the `creates` part prevent overriding this parmeter?

Reply:
You're right that the `creates` is not strictly required.

Though, I think there are perhaps at least 3 relatively solid reasons to keep the `when` clause. To illustrate the thinking, these 3 items are:

(1) expectations, `creates` is a pattern that is part of the shell command to skip a command because the shell command has already been run. In other words, it is typically used to skip a task because that task has already been run, as opposed to some other task was run which makes this task not needed (which is what a `when` is typically used for)

(2) there is a small performance benefit, the `when` clause being false means that ansible will not check the server redundantly for if the file exists

(3) explicit if/else structure between the two tasks marked with a `when`. IE: using a `when: true` and `when: false` blocks creates an if/else structure, using a `when: true` and `creates: file exists` is seemingly mixing structures.